explorer/app/components/Section/Section.tsx

```diff
@@ -1,26 +1,69 @@
-import { Box } from "@mui/material";
-import React from "react";
-import { Text } from "../Text/Text";
+// Core dependencies
+import { Box, Collapse } from "@mui/material";
+import { CollapseProps } from "@mui/material/Collapse/Collapse";
+import React, { ReactNode, useEffect, useState } from "react";
 
-interface SectionProps {
+// App dependencies
+import { CollapseButton } from "./components/CollapseButton/collapseButton";
+import { SectionTitle } from "./components/SectionTitle/sectionTitle";
+import { useBreakpointHelper } from "../../hooks/useBreakpointHelper";
+
+interface Props {
+  children: ReactNode;
+  collapsable?: boolean;
   title?: string;
-  children: React.ReactNode | React.ReactNode[];
 }
 
-export const Section = ({ children, title }: SectionProps): JSX.Element => {
+export const Section = ({
+  children,
+  collapsable = false,
+  title,
+}: Props): JSX.Element => {
+  const mobile = useBreakpointHelper("down", "lg");
+  const [expanded, setExpanded] = useState<boolean>(false);
+  const [transitionDuration, setTransitionDuration] =
+    useState<CollapseProps["timeout"]>(0);
```

Comment:
Can you give me a quick explanation of what CollapseProps is and why we need the delays etc here?